plone/app/contenttypes/browser/link_redirect_view.py

```diff
@@ -111,10 +111,7 @@ def absolute_target_url(self):
             if "resolveuid" in url:
                 uid = url.split("/")[-1]
                 obj = uuidToObject(uid)
-                if obj:
-                    url = "/".join(obj.getPhysicalPath()[2:])
-                    if not url.startswith("/"):
-                        url = "/" + url
+                url = obj.absolute_url()
```

Comment:
I'd check for `obj` first (keep the `if` from line 114) ... it might be that the uuid is missing.

Reply:
also uid = url.split("/")[-1] above is fragile ("/resolveuid/" vs "resolveuid/" vs "/a/relativepath/resolveuid"). Generally, you should find the index of 'resolveuid' in the path, and get the next one if it exists. It should be something like this:
```python
def get_uid(path):
    paths = path.split('/')
    uid = None
    if 'resolveuid' in paths:
        ri = paths.index('resolveuid')
        if ri + 1 != len(paths):
            uid = paths[ri + 1]
            if uid == '':
               uid = None
    return uid 
```
```
>>> get_uid('gfdgdf')
>>> get_uid('')
>>> get_uid('/resolveuid/')
>>> get_uid('/resolveuidgfdgd/')
>>> get_uid('/resolveuid/4342432')
'4342432'
>>> get_uid('/resolveuid/4342432/')
'4342432'
>>> get_uid('fdsfd/fdf/resolveuid/4342432/')
'4342432'
>>> get_uid('/fdsfd/fdf/resolveuid/4342432/')
'4342432'
>>> 
```

Reply:
Also [`/resolveuid/fb91bddde7eb46efbed20e9c10fb4929#autotoc-item-autotoc-1`](https://classic.demo.plone.org/resolveuid/fb91bddde7eb46efbed20e9c10fb4929#autotoc-item-autotoc-1) should still work.

Reply:
@petschki maybe, but i'm not sure

Reply:
At least `plone.outputfilters` still take it into account: https://github.com/plone/plone.outputfilters/blob/master/plone/outputfilters/filters/resolveuid_and_caption.py#L31

Reply:
```python
def get_uid(path):
    paths = path.split('/')
    uid = None
    if 'resolveuid' in paths:
        ri = paths.index('resolveuid')
        if ri + 1 != len(paths):
            uid = paths[ri + 1]
            if '#' in uid:
                uid = uid.split('#')[0]
            if uid == '':
               uid = None
    return uid 
```

```
>>> get_uid('gfdgdf')
>>> get_uid('')
>>> get_uid('/resolveuid/')
>>> get_uid('/resolveuidgfdgd/')
>>> get_uid('/resolveuid/4342432')
'4342432'
>>> get_uid('/resolveuid/4342432/')
'4342432'
>>> get_uid('fdsfd/fdf/resolveuid/4342432/')
'4342432'
>>> get_uid('/fdsfd/fdf/resolveuid/4342432/')
'4342432'
>>> get_uid('/resolveuid/fb91bddde7eb46efbed20e9c10fb4929#autotoc-item-autotoc-1')
'fb91bddde7eb46efbed20e9c10fb4929'
>>> 
```

Reply:
@1letter solution to use `urlparse` is very good! Question: uid's are all lowercase? What if an UID has uppercase letters?

Reply:
@yurj i use a copy of the list with lowercase values https://github.com/plone/plone.app.contenttypes/pull/675/files#diff-c9937c907b85ad438b19cb4ce7abe10606d94a443dc66bef1661a216690f3547R44

Reply:
> At least `plone.outputfilters` still takes it into account: https://github.com/plone/plone.outputfilters/blob/master/plone/outputfilters/filters/resolveuid_and_caption.py#L31

https://github.com/plone/plone.outputfilters/blob/278e2752c4790b5af2470bfc3c91a8aa26a9a80f/plone/outputfilters/filters/resolveuid_and_caption.py#L130

here is what plone.output filter does. It uses the re above to get all parts of the url. If there's an uid, find the object, translate it to the absolute_url, add the other parts.

If we want to do this the same way outputfilter does, we should use this code. This code also check IResolveUidsEnabler. Maybe this last part is not important (if a site disable uids and use resolveuid as a folder name...).

Generally, resoulveuid and catalog path resolution should be done in one place, I think in plone.api. I don't know about circular dipendencies...

Anyway, the actual code is ok, should works as expected and it is a real improvement.